### PR TITLE
[Enhancement II] Compatible processed types

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
Plane segmenations are not compatible with the new `ImagingSpace` here

`RoiResponseSeries` should be fine though since it just has a dynamic table region that can point to anything

### CODE REVIEW

File: spec/ndx-microscopy.extensions.yaml [hunk 1]
```diff
@@ -95,7 +95,7 @@ groups:
       doc: General estimate of location in the brain being subset by this space.
         Specify the area, layer, etc.
         Use standard atlas names for anatomical regions when possible.
-        Specify 'whole-brain' if the entire brain is strictly contained within the space.
+        Specify 'whole brain' if the entire brain is strictly contained within the space.
```
Comment (CodyCBakerPhD) [inline spec/ndx-microscopy.extensions.yaml:98]: Also sneaking this change in

File: spec/ndx-microscopy.extensions.yaml [hunk 2]
```diff
@@ -106,22 +106,14 @@ groups:
     neurodata_type_inc: ImagingSpace
     doc: Metadata about the 2-dimensional slice of physical space that imaging data was recorded from.
     datasets:
-    - name: grid_spacing
+    - name: grid_spacing_in_um
```
Comment (CodyCBakerPhD) [inline spec/ndx-microscopy.extensions.yaml:109]: This was suggested a while back too

I think I like it better

File: spec/ndx-microscopy.extensions.yaml [hunk 3]
```diff
@@ -325,3 +410,79 @@ groups:
       doc: Link to VolumetricImagingSpace object containing metadata about the region of physical space this imaging data
         was recorded from.
       target_type: VolumetricImagingSpace
+
+
+  - neurodata_type_def: VariableDepthMultiChannelMicroscopyVolume
```
Comment (CodyCBakerPhD) [inline spec/ndx-microscopy.extensions.yaml:415]: Also adding a variable-depth static volume

TODO: reduce this block using inheritance

File: spec/ndx-microscopy.extensions.yaml (hunk 1 above)
Comment (CodyCBakerPhD) [reply spec/ndx-microscopy.extensions.yaml:98]: From @alessandratrapani, has not seen the version with the dash, but intuitively wanted 'Whole Brain'

File: spec/ndx-microscopy.extensions.yaml [hunk 4]
```diff
@@ -182,6 +166,106 @@ groups:
...
+      or masks. All segmentation for a given imaging plane is stored together, with
+      storage for multiple imaging planes (masks) supported. Each ROI is stored in its
+      own subgroup, with the ROI group containing both a 2D mask and a list of pixels
+      that make up this mask. Segments can also be used for masking neuropil. If segmentation
+      is allowed to change with time, a new imaging plane (or module) is required and
+      ROI names should remain consistent between them.
+    groups:
+    - neurodata_type_inc: MicroscopyPlaneSegmentation
+      doc: Results from image segmentation of a specific imaging plane.
+      quantity: '+'
```
Comment (CodyCBakerPhD) [inline spec/ndx-microscopy.extensions.yaml:186]: Add 'SummaryImages' here too (mean, correlation, maximum intensity, etc.)

File: spec/ndx-microscopy.extensions.yaml [hunk 5]
```diff
@@ -182,6 +166,106 @@ groups:
       dtype: text
       required: false
 
+
+  # These are needed to allow linkage of processed data to the new objects, until this is merged to core
+  # Technically the RoiResponseSeries shouldn't need to be modified since it just takes a DynamicTableRegion and
+  # does not care about the target
+  - neurodata_type_def: MicroscopyImageSegmentation
```
Comment (CodyCBakerPhD) [inline spec/ndx-microscopy.extensions.yaml:173]: Suggestion from @alessandratrapani, just 'Segmentaiton'

File: spec/ndx-microscopy.extensions.yaml [hunk 6]
```diff
@@ -328,3 +413,79 @@ groups:
...
+      - height
+      - width
+      - depths
+      - channels
+      shape:
+      - null
+      - null
+      - null
+      - null
+    - name: depth_per_frame_in_mm
```
Comment (CodyCBakerPhD) [inline spec/ndx-microscopy.extensions.yaml:467]:
```suggestion
    - name: depth_per_frame_in_um
```

**Comment by CodyCBakerPhD** [comment]
@alessandratrapani I added tests as well as the other items we discussed

**Review by alessandratrapani — APPROVED** [review]
LGTM!

**Pull request merged**